WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -150,6 +150,17 @@ private ICoinsView RemoveNoLock(SmartCoin coin)
 		return coinsToRemove;
 	}
 
+	public bool IsBanned(SmartCoin coin)
```

Comment:
I'm not sure what is being/should be banned. It could be the txo (outpoint) or the script. IMO we should ban the script because coins sharing the same script belong to the same controlling entity.

In case a coin is banned because it is attacking the coordinator the all coins with same script should be prevented from participation because they all belong to the same attacker. 

In case a coin is long banned then, even if other coins sharing the same script are not, the user sooner or later will lose the privacy gained.